nixos/modules/services/hardware/kanata.nix

```diff
@@ -10,7 +10,7 @@ let
       device = mkOption {
         type = types.str;
         example = "/dev/input/by-id/usb-0000_0000-event-kbd";
-        description = "Path to the keyboard device.";
+        description = "Path(s) to the keyboard device.";
```

Comment:
That is not true. This is not a list or a multiline string. I wouldn't know how to add multiple paths here.

Reply:
Linking the upstream doc is fine. We do that a lot.

Reply:
would much prefer if this was a list (and renamed to `devices`). `concatStringsSep ":" devices` is a very small change to the module that makes the interface a lot nicer to use. if our goal was *not* to provide quality of life improvements we wouldn't need to have this option around at all.

Reply:
If we change `device` to `devices` as a list of strs, then path units used to start service units is hard to implement: systemd combines multiple `PathExists` using `or`.

I prefer to only support one device in this module since multiple devices are not easy to implement and not a common use case.

What do you think?

Reply:
Or we can use a heuristic way for path unites: only use `PathExists` for one of the devices.

Because when multi devices is used in one config file, most of the time theses devices are used at the same time.

Reply:
it's reasonable to not support everything a piece of software supports outright in the module. in this case though it might still be possible if each path triggers the next, and the final path then triggers the service.

Reply:
> each path triggers the next

Could you elaborate it a bit? From the manual, path units can only trigger other kinds of units.

I also have tried another idea: `foo.path` and `bar.path` both trigger `foobar.service`, and `foobar.service` has `ConditionPathExists=/path/to/foo` and `ConditionPathExists=/path/to/bar`. This doesn't work because when only `foo` path exits, `foo.path` triggers `foobar.service`, but `foobar.service` fails because `bar` path doesn't exist, then `foo.path` triggers `foobar.service` again and finally reaches rate limit, and `foo.path` is stopped.

Reply:
> Could you elaborate it a bit? From the manual, path units can only trigger other kinds of units.

path units are units. if `a.path` has `Unit=b.path` and `b.path` has `Unit=kanata.service` both conditions must be fulfilled to start kanata. we've verified this works, both if `a.path` triggers first and if `b.path` would trigger first. all it needs to work properly is the service we finally want to trigger to be a oneshot service and to always exit successfully, otherwise it won't be restarted properly.

Reply:
Thanks. The trigger chain is implemented now.

> The argument is a unit name, whose suffix is not ".path".

I was "cheated" by the manual of systemd.path before.